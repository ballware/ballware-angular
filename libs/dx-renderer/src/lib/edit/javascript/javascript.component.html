<div class="dx-texteditor-label" [style.position]="'relative'">{{layoutItem?.options?.caption ?? ''}}</div>
<ballware-codemirror 
    mode="javascript"
    [readOnly]="readonly$ | async"
    width="100%"
    [height]="height ?? '300px'"
    [options]="options"
    [(value)]="value">
</ballware-codemirror>    