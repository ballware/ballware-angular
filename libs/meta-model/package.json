{
  "name": "@ballware/meta-model",
  "version": "8.0.1",
  "type": "commonjs"
}
