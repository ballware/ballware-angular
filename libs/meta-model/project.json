{
  "name": "meta-model",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/meta-model/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": [
        "{options.outputPath}"
      ],
      "options": {
        "outputPath": "dist/libs/meta-model",
        "main": "libs/meta-model/src/index.ts",
        "tsConfig": "libs/meta-model/tsconfig.lib.json",
        "assets": [
          "libs/meta-model/*.md"
        ]
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/publish.mjs meta-model {args.ver} {args.tag}"
      },
      "dependsOn": [
        "build"
      ]
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": [
        "{options.outputFile}"
      ],
      "options": {
        "lintFilePatterns": [
          "libs/meta-model/**/*.ts"
        ]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": [
        "{workspaceRoot}/coverage/{projectRoot}"
      ],
      "options": {
        "jestConfig": "libs/meta-model/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
