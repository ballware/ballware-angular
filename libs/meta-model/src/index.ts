export * from './lib/cruditem';
export * from './lib/document';
export * from './lib/entity';
export * from './lib/page';
export * from './lib/queryparams';
export * from './lib/rights';
export * from './lib/scriptutil';
export * from './lib/statistic';
export * from './lib/template';
export * from './lib/tenant';

