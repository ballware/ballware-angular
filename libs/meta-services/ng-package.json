{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/libs/meta-services",
  "lib": {
    "entryFile": "src/index.ts"
  }
}
