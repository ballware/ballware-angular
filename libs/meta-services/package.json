{
  "name": "@ballware/meta-services",
  "version": "8.0.6",
  "peerDependencies": {
    "@angular/common": "^14.2.0",
    "@angular/core": "^14.2.0"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  }
}
