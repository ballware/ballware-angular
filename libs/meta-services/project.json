{
  "name": "meta-services",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/meta-services/src",
  "prefix": "ballware",
  "targets": {
    "build": {
      "executor": "@nrwl/angular:package",
      "outputs": [
        "dist/libs/meta-services"
      ],
      "options": {
        "project": "libs/meta-services/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/meta-services/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/meta-services/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": [
        "coverage/libs/meta-services"
      ],
      "options": {
        "jestConfig": "libs/meta-services/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/meta-services/**/*.ts",
          "libs/meta-services/**/*.html"
        ]
      }
    }
  },
  "tags": []
}
