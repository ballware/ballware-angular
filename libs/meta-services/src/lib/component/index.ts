import { StoreModule } from '@ngrx/store';
import { componentReducer } from './component.reducer';
import { componentFeatureKey } from './component.state';

export * from './component.actions';
export * from './component.state';

export const ComponentFeatureModule = StoreModule.forFeature(componentFeatureKey, componentReducer);