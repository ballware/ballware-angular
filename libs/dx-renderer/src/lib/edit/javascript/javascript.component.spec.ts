import { ComponentFixture, TestBed } from '@angular/core/testing';

import { EditLayoutJavascriptComponent } from './javascript.component';
import { Provider } from '@angular/core';
import { EDIT_SERVICE } from '@ballware/meta-services';
import { EditLayoutItem } from '@ballware/meta-model';
import { mockedEditServiceContext } from '../../../test/editservice.spec';

describe('EditLayoutJavascriptComponent', () => {
  let component: EditLayoutJavascriptComponent;
  let fixture: ComponentFixture<EditLayoutJavascriptComponent>;

  const mockedEditService = mockedEditServiceContext();
        
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [ EditLayoutJavascriptComponent ],
      providers: [        
        {
          provide: EDIT_SERVICE,
          useFactory: () => mockedEditService.mock.object()
        } as Provider         
      ]
    })
    .compileComponents();
  });

  it('should create', () => {
    fixture = TestBed.createComponent(EditLayoutJavascriptComponent);

    const layoutItem = {
      type: 'text',
      options: {
          dataMember: 'mockedmember',
      }
    } as EditLayoutItem;
    
    component = fixture.componentInstance;   
    expect(component).toBeTruthy();

    component.initialLayoutItem = layoutItem;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should apply options', () => {
    fixture = TestBed.createComponent(EditLayoutJavascriptComponent);

    const layoutItem = {
      options: {
          dataMember: 'mockedmember',
          required: false,
          readonly: false,            
          visible: false
      }
    } as EditLayoutItem;
    
    component = fixture.componentInstance;   
    expect(component).toBeTruthy();

    component.initialLayoutItem = layoutItem;
    fixture.detectChanges();

    expect(component.getOption('value')).toBe("");
    expect(component.getOption('required')).toBe(false);
    expect(component.getOption('readonly')).toBe(false);
    expect(component.getOption('visible')).toBe(false);

    component.setOption('value', 'some text');
    expect(component.getOption('value')).toBe('some text');

    component.setOption('required', true);
    expect(component.getOption('required')).toBe(true);

    component.setOption('readonly', true);
    expect(component.getOption('readonly')).toBe(true);

    component.setOption('visible', true);
    expect(component.getOption('visible')).toBe(true);

    expect(() => component.getOption('undefined')).toThrowError('Unsupported option <undefined>');
    expect(() => component.setOption('undefined', 'any value')).toThrowError('Unsupported option <undefined>');
  });
});
