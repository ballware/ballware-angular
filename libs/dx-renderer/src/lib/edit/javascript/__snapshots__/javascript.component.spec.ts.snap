// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditLayoutJavascriptComponent should create 1`] = `
<ballware-edit-javascript
  _optionRegistry={[Function Array]}
  currentValue$={[Function BehaviorSubject]}
  destroy$={[Function Subject]}
  editService={[Function ]}
  editorEntered$={[Function Subject]}
  editorEvent$={[Function Subject]}
  emailValidation$={[Function BehaviorSubject]}
  getOption={[Function Function]}
  globalReadonly="false"
  height="undefined"
  initialLayoutItem={[Function Object]}
  layoutItem={[Function Object]}
  notifyValueChange$={[Function Subject]}
  onEntered={[Function Function]}
  onEvent={[Function Function]}
  options="undefined"
  preparedLayoutItem$={[Function BehaviorSubject]}
  readonly$={[Function BehaviorSubject]}
  refreshValueTrigger$={[Function Subject]}
  registerOption={[Function Function]}
  required$={[Function BehaviorSubject]}
  requiredValidation$={[Function BehaviorSubject]}
  setOption={[Function Function]}
  validationRules$={[Function Observable]}
  visible$={[Function BehaviorSubject]}
>
  <div
    class="dx-texteditor-label mt-2"
    style="position: relative;"
  >
  </div><ballware-codemirror
    mode="javascript"
    width="100%"
  >
    <div
      style="width: 100%; height: 300px;"
    />
  </ballware-codemirror>
</ballware-edit-javascript>
`;
